compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/GenerateClosuresForDefaultValues.java

```diff
@@ -524,6 +545,25 @@ private BLangExpression createClosureForDefaultValue(String closureName, String
         env.enclPkg.functions.add(function);
         env.enclPkg.topLevelNodes.add(function);
         symbol.defaultValues.put(paramName, varSymbol);
+        rewrite(lambdaFunction, lambdaFunction.capturedClosureEnv);
+        return returnStmt.expr;
+    }
+
+    private BLangExpression createClosureForDefaultValueOfRecordField(String closureName, String paramName,
+                                                                      BLangSimpleVariable varNode) {
+        BSymbol owner = getOwner(env);
+        BRecordTypeSymbol symbol = (BRecordTypeSymbol) env.node.getBType().tsymbol;
+        BLangFunction function = createFunction(closureName, varNode.pos, symbol.pkgID, owner, varNode.getBType());
+        BLangReturn returnStmt = ASTBuilderUtil.createReturnStmt(function.pos, (BLangBlockFunctionBody) function.body);
+        returnStmt.expr = varNode.expr;
+        BLangLambdaFunction lambdaFunction = createLambdaFunction(function);
+        lambdaFunction.capturedClosureEnv = env.createClone();
+        BInvokableSymbol varSymbol = createSimpleVariable(function, lambdaFunction);
+        env.enclPkg.symbol.scope.define(function.symbol.name, function.symbol);
+        env.enclPkg.functions.add(function);
+        env.enclPkg.topLevelNodes.add(function);
+        symbol.defaultValues.put(paramName, varSymbol);
+        rewrite(lambdaFunction, lambdaFunction.capturedClosureEnv);
         return returnStmt.expr;
```

Comment:
Isn't this basically the original expression? Why do we need to return it?

Reply:
removed